### DISCUSSION TITLE
Ollama Support?

### DISCUSSION THREAD

**Discussion by rhyswynn** [SakanaAI/AI-Scientist#146] [Ideas]
I know the recommendation is for frontier models, but fine tuned smaller models could work well and offer more cost efficient options. Any plans for adding ollama and hugging face options?

**Comment by BradKML** [comment]
This but OpenRouter (see also https://github.com/SakanaAI/AI-Scientist/discussions/59)

**Comment by aipdv** [comment]
see #98 for using ollama. infact i've been onto integrating other models like **qwq, qwen2.5, deepseek via hyperbolic platform**.

**Reply by BradKML** [reply]
Noted with thanks, would love to see the same flexibility with OpenRouter for niche larger models.
P.S. do you know how to get some of the extra niche HF models to work on OpenRouter? Ollama can support then locally if the GPU is beefy enough but otherwise there has got to be another way
